R/runs.R

```diff
@@ -14,14 +14,33 @@ metric_value_to_rest <- function(value) {
   }
 }
 
-#' Log Metric
+#' @importFrom tibble tibble
+#' @importFrom rlang names2
+get_key_value_df <- function(...) {
+  values <- list(...) %>% unlist() %>% map_chr(metric_value_to_rest)
```

Comment:
Do we _always_ want to do this? I'm pretty sure metrics are numbers but params are chars

Reply:
we could def skip this for params. i just don't know what the best way to implement that is. do we pass `"metric"` or `"params"` (or any kind of indicator really) to this function, and skip the value coercion if params?

Reply:
🤮  yeah that's brutal. the only other thing that comes to mind is to get the name of the next call up in the call stack (i.e. the function that called this function) and branch based on that, but that seems really brittle

Reply:
eh, it is is sort of brittle, but also kind of elegant? we would never expect the user to call this function, so i feel like we can justify the hackiness of it

Reply:
worth a shot. I don't think it'll be pipeable that way though?